### ISSUE TITLE
Client ID 21 - Update Layer Title and Description

### ISSUE THREAD

**Issue by tdipisa** [geosolutions-it/austrocontrol-C125#243]
(f) when layer name is changed, title in MS is replaced by original layer name and description is removed. There is no warning!
E.g. Create Layer ""AD"" with ""Title Aerodrome"" in GeoServer  -  Add it to MapStore  -  Change Layer name in GeoServer to ""AD Test""  -  Update Layer in MapStore --> MapStore sets the layer title to ""AD"" and removes all other infoamtion without any warning.
Expected behaviour: MapStore should warn the administrator, that the updated layer is not available anymore. All currently available information should stay as it is.

- Layers are sorted in oppsite order of ToC - is it possible to order alphabetically or implement a search bar to make it easier for the admin to find individual layers?

**Comment by tdipisa** [comment]
> (f) when layer name is changed, title in MS is replaced by original layer name and description is removed. There is no warning!
E.g. Create Layer ""AD"" with ""Title Aerodrome"" in GeoServer - Add it to MapStore - Change Layer name in GeoServer to ""AD Test"" - Update Layer in MapStore --> MapStore sets the layer title to ""AD"" and removes all other infoamtion without any warning.

>Expected behaviour: MapStore should warn the administrator, that the updated layer is not available anymore. All currently available information should stay as it is.

We have to check it thoroughly. There is probably a regression to fix.

>Layers are sorted in oppsite order of ToC - is it possible to order alphabetically or implement a search bar to make it easier for the admin to find individual layers?

We can ofcourse fix the sorting of layers by respecting the one in TOC. We will evaluate the cost of a search bar even though it is not a requirement and we will let you know.

**Comment by tdipisa** [comment]
@michegg

> (f) when layer name is changed, title in MS is replaced by original layer name and description is removed. There is no warning!
E.g. Create Layer ""AD"" with ""Title Aerodrome"" in GeoServer - Add it to MapStore - Change Layer name in GeoServer to ""AD Test"" - Update Layer in MapStore --> MapStore sets the layer title to ""AD"" and removes all other infoamtion without any warning.

>Expected behaviour: MapStore should warn the administrator, that the updated layer is not available anymore. All currently available information should stay as it is.

I've checked and everything seems to be working as expected for me, see the GIF below.

![update_layer](https://user-images.githubusercontent.com/1280027/100208553-2496b600-2f09-11eb-9feb-525a7d86f265.gif)

**Comment by michegg** [comment]
I get a different behavior when using your installation at [https://austrocontrol.geo-solutions.it/austrocontrol-ms2](https://austrocontrol.geo-solutions.it/austrocontrol-ms2) 
Sorry for the bad quality. I hope it's visible in the GIF below, that the layer is "updated successfully", but the title and description are removed, while the title for ToC entry name is changed to the previous layer name.
Layer is marked by a red !, when zooming in or out after the update.
Zooming in or out before the update: layer is also marked by a red !, but behavior of update process is the same (no warning).

I get the same result, when using other maps on the installation.

![update_layer](https://user-images.githubusercontent.com/57953447/100235401-ab10bf00-2f2c-11eb-9c7b-8a57f0967915.gif)

**Comment by tdipisa** [comment]
@michegg it is strange. Can you please clean your browser cache and try again?

**Comment by michegg** [comment]
I get the same result after cleaning the cache. I'm using Chrome and also tried Firefox.
I'm only changing the layer name in GeoServer, while you also modify title and description.
Maybe that's why we are getting different results?

**Comment by tdipisa** [comment]
Dear @michegg 

> I'm only changing the layer name in GeoServer, while you also modify title and description.
> Maybe that's why we are getting different results?

I've also tested also by changing only the layer name to be honest and it is working fine for me.

I've put this issue in test again since we have implemented the modification you requested to have the layer sorted as in TOC. Therefore I will ask testers to do more testing. If testers will not be able to replicate what you are reported, the only way I see now is to schedule a call to check it togheter with you.

**Comment by tdipisa** [comment]
@ElenaGallo what about my last comment and the revious one reported by @michegg?

Can you provide feedback please before closing the issue? The issue cannot be closed directly, we have to understand why the client is experiencing the mentioned problem.

**Comment by ElenaGallo** [comment]
@tdipisa @michegg I tested the issues by both changing only the name of the layer, and also changing the title and description (as reported in the comments above) but I am not able to replicate the problem either.

**Comment by tdipisa** [comment]
@michegg if you want we can schedule a brief call to check it together.

**Comment by michegg** [comment]
@tdipisa: I'm available today from 13:30, tomorrow until 12:00 or on friday.
Please contact me via mail and let me know which technology to use.

**Comment by tdipisa** [comment]
After the today meeting with @michegg, we found that the problem occurs when the base URL of the catalog source contains the layer's workspace. 

Tested again with the following URL in the Catalog tool:
http://gs-stable.geo-solutions.it/geoserver/topp/wms

and adding this layer to the map:
states_test

When you change the layer metadata in GeoServer (like name, title and description) and then you try to update them MS side through the update tool, MS reports a success even if the update requests should fails and layer metadata are wrongly updated in Layer Settings (they are empty). If then you manually update the layer name in MS and run again the update tool, layer metadata are properly updated as expected: probably we just need to enforce the check during the GetCapabilities response parsing in this particular use case while the workspace is present in the GetCapabilities request URL.

Below is a GIF to replicate the issue in the [austrocontrol DEV instance](https://austrocontrol.geo-solutions.it/austrocontrol-ms2/)

![update_layer](https://user-images.githubusercontent.com/1280027/101899685-82382d00-3bae-11eb-8595-13e5b0187618.gif)
